### DISCUSSION TITLE
Can the SQLite database be extended (with SpatiaLite)?

### DISCUSSION THREAD

**Discussion by maarten-22** [pocketbase/pocketbase#3902] [Q&A] [answered]
Out of curiosity, can the SQLite database be extended? For example with SpatiaLite to allow for spatial SQL capabilities? If someone has some information about this, I'd like to read some about it :)

**Comment by ganigeorgiev** [comment, marked as answer]
It is not possible at the moment to load custom extensions or SQLite build without forking the project and replacing the [`db_cgo/connectDB()`](https://github.com/pocketbase/pocketbase/blob/master/core/db_cgo.go) function (_there is also an option to use the `go:linkname` directive to replace the function during compilation but outside of tests I don't recommend it as it is possible to break something else_).

There are plans in the future to allow more "native" SQLite dev experience that will allow customizing the above but this is temporary pushed aside for now as other tasks are with higher priority.

**Reply by maarten-22** [reply]
Hey, thanks for letting me know! I'll use something else more suited for now then. I might migrate back to PB in the future :). Good luck on everything!

**Reply by basaran** [reply]
Arigato, I was able to compile pocketbase with spatialite support:

```golang
	sql.Register("pb_sqlite3",
		&sqlite3.SQLiteDriver{
			Extensions: []string{"mod_spatialite"}, // here
			ConnectHook: func(conn *sqlite3.SQLiteConn) error {
				_, err := conn.Exec(`
					PRAGMA busy_timeout       = 10000;
					PRAGMA journal_mode       = WAL;
					PRAGMA journal_size_limit = 200000000;
					PRAGMA synchronous        = NORMAL;
					PRAGMA foreign_keys       = ON;
					PRAGMA temp_store         = MEMORY;
					PRAGMA cache_size         = -16000;
				`, nil)

				return err
			},
		},
	)

```

I confirmed the new binary has access to the spatial functions. However, you would still need to install the spatialite library to your OS so sqlite can find it from your `/usr/lib` or whatever.
